org.eclipse.winery.tools.deployablecomponents/src/main/java/org/eclipse/winery/tools/deployablecomponents/crawler/GithubCrawler.java

```diff
@@ -0,0 +1,232 @@
+        String responseString = HttpGet(url, true, false);
+
+        JSONArray responseArray = new JSONArray(responseString);
+        JSONObject response = responseArray.getJSONObject(0);
+        lastId = response.getInt("id");
+
+        return parseJsonToRepository(response);
+    }
+
+    private Repository parseJsonToRepository(JSONObject json) throws MalformedURLException, JSONException {
+        String name = json.getString("full_name");
+        String urlString = json.getString("url");
+
+        URL url = new URL(urlString);
+
+        return new Repository(name, url);
+    }
+
+    private String HttpGet(URL url, boolean isInRateLimit, boolean isSearch) throws IOException {
+        if (isInRateLimit && isSearch && rateLimitRemainingSearch <= 0) {
+            sleepSearchRateLimit();
+        }
+        if (isInRateLimit && rateLimitRemaining <= 0) {
+            sleepRateLimit();
+        }
+
+        boolean successful;
+        StringBuilder responseBuilder;
+        HttpURLConnection connection;
+        int fails = 0;
+        // as long as rate limit is not exceeded this do-while is done only once
+        do {
+            if (fails >= 10) {
+                throw new IOException("HTTP request failed " + fails + " times. Request canceled!");
+            }
+            responseBuilder = new StringBuilder();
+            connection = (HttpURLConnection) url.openConnection();
+            connection.setRequestMethod("GET");
+            connection.setDoOutput(true);
+            if (authenticated) {
+                connection.setRequestProperty("Authorization", "Basic " + authEncoded);
+            }
+
+            if (connection.getResponseCode() != HttpURLConnection.HTTP_OK) {
+                // 403 Forbidden might be rate limit
+                if (connection.getResponseCode() == HttpURLConnection.HTTP_FORBIDDEN) {
+                    String limitRemaining = connection.getHeaderField("X-RateLimit-Remaining");
+                    if (limitRemaining != null && Integer.parseInt(limitRemaining) == 0) {
+                        successful = false;
+                        if (isSearch) {
+                            rateLimitResetSearch = Instant.ofEpochSecond(Long.parseLong(connection.getHeaderField("X-RateLimit-Reset")));
+                            connection.disconnect();
+                            sleepSearchRateLimit();
+                        } else {
+                            rateLimitReset = Instant.ofEpochSecond(Long.parseLong(connection.getHeaderField("X-RateLimit-Reset")));
+                            connection.disconnect();
+                            sleepRateLimit();
+                        }
+                    } else {
+                        fails++;
+                        successful = false;
+                        System.err.println("An error occurred in a http get request: " + connection.getResponseCode() + " - " + connection.getResponseMessage() + ". Failed " + fails + ". time.");
```

Comment:
Please use the SLF4J Logger instead of System to log info and errors.

This holds for the whole code, so please adjust all code accordingly.